catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactory.java

```diff
@@ -232,6 +233,90 @@ private Filter transformCustomFunctionToFilter(Filter filter) {
         return filter;
     }
 
+    private QueryRequest normalizeSort(QueryRequest request) {
+        if (request == null) {
+            return null;
+        }
+
+        Query query = request.getQuery();
+        if (query == null) {
+            return request;
+        }
+
+        List<SortBy> sortBys = new ArrayList<>();
+
+        // Primary sort parameter
+        SortBy sortBy = query.getSortBy();
+        SortBy newSortBy = normalizeSortBy(sortBy);
+        if (newSortBy != null) {
+            sortBys.add(newSortBy);
+        }
+
+        // Additional sort parameters
+        Serializable extraSortBys = request.getPropertyValue(EXT_SORT_BY);
+        if (extraSortBys != null && extraSortBys instanceof SortBy[]) {
+            List<SortBy> extraSortBysList = Arrays.asList((SortBy[]) extraSortBys);
+            extraSortBysList.stream()
+                    .map(this::normalizeSortBy)
+                    .filter(Objects::nonNull)
+                    .forEach(sortBys::add);
+        }
+
+        Map<String, Serializable> newProperties = request.getProperties();
+        SortBy normalizedSortBy = null;
+        if (sortBys.size() > 0) {
+            normalizedSortBy = sortBys.get(0);
+        }
+        if (sortBys.size() > 1) {
+            SortBy[] extraSortBysArray = sortBys.subList(1, sortBys.size())
+                    .toArray(new SortBy[0]);
+            newProperties.put(EXT_SORT_BY, extraSortBysArray);
```

Comment:
what happens if `newProperties` is null at this point?

Reply:
If `newProperties` is null then `sortBys.size()` will never be greater than 1 so this `put()` will never happen. The `remove()` 2 lines down though will blow up. I refactored the logic to account for a null `newProperties`.